### DISCUSSION TITLE
IdeaVIM leaves Visual mode when pressing arrow keys on JetBrains Gateway

### DISCUSSION THREAD

**Discussion by docent** [JetBrains/ideavim#849] [Q&A] [answered]
Hi,
first of all don't blame me for using arrows, I've been doing that forever and hard to switch now ;) I've been experiencing a problem with visual mode when using JetBrains Gateway. My setup is the following:
Mac M2 -> Jetbrains Gateway 2024.1 Beta -> Idea 2024.1 Beta (running on a RH VM, need Idea Beta for Jetty 12 support...). IdeaVIM 2.10.0 plugin only installed on the client.

Currently when I enter a visual select, whether by line or by character, it exits as soon as I press any arrow key! Also, when I press "down arrow" while in visual it jumps down two lines.

Interestingly I cannot reproduce this issue when I change the backend (I fired a rustrover project and it doesn't happen), and doesn't happen in a locally installed Intellij Idea. It also worked somehow before the plugin updated to 2.10.0 (but not always). Generally it's hard to say what triggers this bug exactly.

To be sure, I have:
- IdeaVIM installed only on the client
- no custom .ideavimrc
- Tried to reset settings
- Tried to create a new project

Found an old issue someone had...
https://youtrack.jetbrains.com/issue/GTW-2100/IdeaVim-Text-selection-in-the-Visual-mode-doesnt-work-by-arrows
Any assistance would be appreciated, as it's driving me nuts :)

**Comment by AlexPl292** [comment, marked as answer]
Hi there! Can you please take a look if 2.10.1 patch fixes your problem? We discovered a regression with removed dev and 2.10.0, probably you're also affected by it.

**Reply by docent** [reply]
Hey Alex, yeah, updating to 2.10.1 solved it. Thanks!
